pkg/image/containerdregistry/registry.go

```diff
@@ -100,13 +100,13 @@ func (r *Registry) Unpack(ctx context.Context, ref image.Reference, dir string)
 		return err
 	}
 
-	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
+	if err := os.MkdirAll(to, os.ModePerm); err != nil {
 		return err
 	}
 
 	for _, layer := range manifest.Layers {
 		r.log.Debugf("unpacking layer: %v", layer)
```

Comment:
Looks like `from` isn't being respected yet?

Reply:
Yes, in the containerd case, since you're unpacking by layer after looking at the manifest file, I haven't added the `from` since in this case unpacking the entire filesystem makes sense (and the edge cases are resolved). I'm not sure we want to change that behavior in PR necessarily, since the goal is to fix the issue in unpacking when opm shells out to external tools

Reply:
I think this **needs** to respect the interface's contract and only copy the files/directories on the path specified with the `from` argument. It would be very odd if this implementation ignores the `from` argument.

The `unpackLayer` method can filter out the paths that get copied, so this may just amount to adding another filter there.

Reply:
looks like this was fixed! nice